pkg/kubectl/cmd/top_node.go

```diff
@@ -103,5 +111,23 @@ func (o TopNodeOptions) RunTopNode() error {
 	if err != nil {
 		return err
 	}
-	return o.Printer.PrintNodeMetrics(metrics)
+
+	selector := labels.Everything()
+	if len(o.Selector) > 0 {
+		selector, err = labels.Parse(o.Selector)
+		if err != nil {
+			return err
+		}
+	}
+	nodes, err := o.Client.Nodes().List(api.ListOptions{
```

Comment:
You're ignoring this error.  I'm good with graceful degradation, but I still think the error should be reported on stderr.